contracts/pcv/PCVGuardian.sol

```diff
@@ -0,0 +1,181 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity ^0.8.4;
+
+import {EnumerableSet} from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+import {CoreRef} from "../refs/CoreRef.sol";
+import {IPCVGuardian} from "./IPCVGuardian.sol";
+import {IPCVDeposit} from "./IPCVDeposit.sol";
+import {CoreRefPauseableLib} from "../libs/CoreRefPauseableLib.sol";
+import {TribeRoles} from "../core/TribeRoles.sol";
+
+/// @notice PCV Guardian is a contract to safeguard protocol funds
+/// by being able to withdraw whitelisted PCV deposits to an immutable safe address
+contract PCVGuardian is IPCVGuardian, CoreRef {
+    using CoreRefPauseableLib for address;
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    ///@notice set of whitelisted pcvDeposit addresses for withdrawal
+    EnumerableSet.AddressSet private whitelistAddresses;
+
+    ///@notice safe address where PCV funds can be withdrawn to
+    address public immutable safeAddress;
+
+    constructor(
+        address _core,
+        address _safeAddress,
+        address[] memory _whitelistAddresses
+    ) CoreRef(_core) {
+        safeAddress = _safeAddress;
+
+        // improbable to ever overflow
+        unchecked {
+            for (uint256 i = 0; i < _whitelistAddresses.length; i++) {
+                _addWhitelistAddress(_whitelistAddresses[i]);
+            }
+        }
+    }
+
+    modifier onlyWhitelist(address pcvDeposit) {
+        require(
+            isWhitelistAddress(pcvDeposit),
+            "Provided address is not whitelisted"
+        );
+        _;
+    }
+
+    // ---------- Read-Only API ----------
+
+    /// @notice returns true if the the provided address is a valid destination to withdraw funds from
+    /// @param pcvDeposit the pcvDeposit address to check if whitelisted
+    function isWhitelistAddress(address pcvDeposit)
+        public
+        view
+        override
+        returns (bool)
+    {
+        return whitelistAddresses.contains(pcvDeposit);
+    }
+
+    /// @notice returns all whitelisted pcvDeposit addresses
+    function getWhitelistAddresses()
+        public
+        view
+        override
+        returns (address[] memory)
+    {
+        return whitelistAddresses.values();
+    }
+
+    // ---------- Governor-Only State-Changing API ----------
+
+    /// @notice governor-only method to whitelist a pcvDesposit address to withdraw funds from
+    /// @param pcvDeposit the address to set as safe
```

Comment:
Natspec comment I think is out of date. The `pcvDeposit` in this implementation isn't a safe address - you're whitelisting it to be eligible to be withdrawn from